src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -803,7 +810,13 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
                 project.getRemoteProjectRepositories(),
                 getReactorModelsFromSession());
         Properties userProperties = this.session.getUserProperties();
+        settings.getProfiles()
+                .stream()
+                .filter(p -> settings.getActiveProfiles().contains(p.getId()))
+                .forEach(activeProfile -> userProperties.putAll(activeProfile.getProperties()));
+
         List<String> activeProfiles = this.session.getRequest().getActiveProfiles();
+        activeProfiles.addAll(this.settings.getActiveProfiles());
```

Comment:
I'm not sure if we try to modify list returned from request.
In Maven 4 most of objects will be immutable.

Reply:
I created local var instead of modifying the ones from the Maven objects